src/main/scala/io/iohk/ethereum/forkid/ForkId.scala

```diff
@@ -0,0 +1,78 @@
+package io.iohk.ethereum.forkid
+
+import java.util.zip.CRC32
+import java.nio.ByteBuffer
+
+import akka.util.ByteString
+import io.iohk.ethereum.utils.BlockchainConfig
+import io.iohk.ethereum.utils.BigIntExtensionMethods._
+import io.iohk.ethereum.utils.ByteUtils._
+import io.iohk.ethereum.utils.Hex
+import io.iohk.ethereum.rlp._
+
+import RLPImplicitConversions._
+
+case class ForkId(hash: BigInt, next: Option[BigInt]) {
+  override def toString(): String = s"ForkId(0x${Hex.toHexString(hash.toUnsignedByteArray)}, $next)"
+}
+
+object ForkId {
+
+  def create(genesisHash: ByteString, config: BlockchainConfig)(head: BigInt): ForkId = {
+    val crc = new CRC32()
+    crc.update(genesisHash.asByteBuffer)
+    val next = gatherForks(config).find { fork =>
+      if (fork <= head) {
+        crc.update(bigIntToBytes(fork, 8))
+      }
+      fork > head
+    }
+    new ForkId(crc.getValue(), next)
+  }
+
+  val noFork = BigInt("1000000000000000000")
+
+  def gatherForks(config: BlockchainConfig): List[BigInt] = {
+    val maybeDaoBlock: Option[BigInt] = config.daoForkConfig.flatMap { daoConf =>
+      if (daoConf.includeOnForkIdList) Some(daoConf.forkBlockNumber)
+      else None
+    }
+
+    (maybeDaoBlock.toList ++
+      config.forkBlockNumbers.productIterator.toList.flatMap {
+        case i: BigInt => Some(i)
+        case i: Option[_] =>
+          i.flatMap {
+            case n if n.isInstanceOf[BigInt] => Some(n.asInstanceOf[BigInt])
+            case n => None
+          }
+        case default => None
+      })
+      .filterNot(v => v == 0 || v == noFork)
```

Comment:
I don't get why we filter out `noFork`. It does not seem to be used somewhere else

Reply:
The same configuration structure `BlockchainConfig` is re-used for multiple chains. Forks which shouldn't occur in a give chain are configured to happen at block height "1000000000000000000" which is equivalent to "never". We don't want those fork numbers on our fork id list